File: JavaScript/8js 1/check_nl2.js
/*
|--------------------------------------------------------------------------
| No Loops 2 - You only need one
|--------------------------------------------------------------------------
|
*/

function check(a, x) {
   return a.includes(x);
}


function check(a, x) {
   return a.indexOf(x) >= 0;
}


function check(a, x) {
   return a.some((v) => v == x);
}


console.log(check([66, 101], 66), true);
console.log(check([80, 117, 115, 104, 45, 85, 112, 115], 45), true);
console.log(check(["t", "e", "s", "t"], "e"), true);
console.log(check(["what", "a", "great", "kata"], "kat"), false);

File: Python/8py 1/check_nl2.py
"""
|--------------------------------------------------------------------------
| No Loops 2 - You only need one
|--------------------------------------------------------------------------
|
"""

def check(a, x):
   return x in a


check = list.__contains__
from operator import contains as check


def check(a, x): 
   return a.count(x) > 0


print(check([66, 101], 66), True)
print(check([80, 117, 115, 104, 45, 85, 112, 115], 45), True)
print(check(['t', 'e', 's', 't'], 'e'), True)
print(check(['what', 'a', 'great', 'kata'], 'kat'), False)